Source/Scene/modelMaterialsCommon.js

```diff
@@ -0,0 +1,792 @@
+                fragmentShader += '  vec3 ambient = texture2D(u_ambient, ' + v_texcoord + ');\n';
+            }
+            else {
+                fragmentShader += '  vec3 ambient = u_ambient.rgb;\n';
+            }
+        }
+        else {
+            fragmentShader += '  vec3 ambient = diffuse.rgb;\n';
+        }
+        fragmentShader += '  vec3 viewDir = -normalize(v_positionEC);\n';
+        fragmentShader += '  vec3 ambientLight = vec3(0.0, 0.0, 0.0);\n';
+        colorCreationBlock += '  color += ambient * ambientLight;\n';
+
+        // Add in light computations
+        fragmentShader += fragmentLightingBlock;
+
+        fragmentShader += colorCreationBlock;
+        fragmentShader += finalColorComputation;
+        fragmentShader += '}\n';
+
+        // TODO: Handle texture transparency
+        var techniqueStates;
+        if (hasAlpha) {
+            techniqueStates = {
+                enable: [
+                    WebGLConstants.DEPTH_TEST,
+                    WebGLConstants.BLEND
+                ],
+                depthMask: false,
+                functions: {
+                    "blendEquationSeparate": [
+                        WebGLConstants.FUNC_ADD,
+                        WebGLConstants.FUNC_ADD
+                    ],
+                    "blendFuncSeparate": [
+                        WebGLConstants.ONE,
+                        WebGLConstants.ONE_MINUS_SRC_ALPHA,
+                        WebGLConstants.ONE,
+                        WebGLConstants.ONE_MINUS_SRC_ALPHA
+                    ]
+                }
+            };
+        }
+        else {
+            techniqueStates = {
+                enable: [
+                    WebGLConstants.CULL_FACE,
+                    WebGLConstants.DEPTH_TEST
+                ]
+            };
+        }
+        techniques[techniqueId] = {
+            attributes: techniqueAttributes,
+            parameters: techniqueParameters,
+            program: programId,
+            states: techniqueStates,
+            uniforms: techniqueUniforms
+        };
+
+        // Add shaders
+        shaders[vertexShaderId] = {
+            type: WebGLConstants.VERTEX_SHADER,
+            uri: '',
+            extras: {
+                source: vertexShader
+            }
+        };
+        shaders[fragmentShaderId] = {
+            type: WebGLConstants.FRAGMENT_SHADER,
+            uri: '',
+            extras: {
+                source: fragmentShader
+            }
+        };
+
+        // Add program
+        var programAttributes = [];
+        for (i=0;i<attributesCount;++i) {
+            programAttributes.push('a_' + attributes[i].toLowerCase());
+        }
+
+        programs[programId] = {
+            attributes: programAttributes,
+            fragmentShader: fragmentShaderId,
+            vertexShader: vertexShaderId
+        };
+
+        return techniqueId;
+    }
+
+    function getKey(khrMaterialsCommon) {
```

Comment:
These technique ids could get really long, right?  This is OK-ish for runtime use, but what if this is used as part of the content pipeline?

Perhaps we should use names like `common-material-0`, `common-material-1`, etc., and just use this string as a hash to look these up while doing the conversion.

Also, we need to make sure the asset doesn't already have a technique with the same name since an asset can mix stock glTF techniques and KHR_common_materials